### PR TITLE
Lock around variables used in Update()

### PR DESCRIPTION
**Pull request by stuartnelson3**
Found two places where struct members being
updated in api.Update() where being accessed
elsewhere without locks.

testing this at the moment.

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -331,6 +331,7 @@ func (api *API) listAlerts(w http.ResponseWriter, r *http.Request) {
 	alerts := api.alerts.GetPending()
 	defer alerts.Close()
 
+	api.mtx.RLock()
```
Comment (brancz) [inline api/api.go]: What are we locking against here?

Comment (stuartnelson3) [reply api/api.go]: there's a `routes := api.route.Match(a.Labels)` a few lines below, in a loop. This is one of the fields that gets updated in `Update()`

Comment (brancz) [reply api/api.go]: Got it. I missed that.

**Comment by brancz** [comment]
lgtm

File: api/api.go (hunk 1 above)
Comment (stuartnelson3) [reply api/api.go]: no worries, github didn't bother showing it 🤷‍♂️

Comment (brancz) [reply api/api.go]: I actually expanded but missed the `api.route` part. So it's just me needing coffee.

**Pull request merged**